### ISSUE TITLE
✅ 🐞 Bug Report: Menu Widget 'Fit to Content' Dropdown x position calculations are incorrect after v3.16.0 update.

### ISSUE THREAD

**Issue by lizsterine** [elementor/elementor#23808]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

The latest updates to Elementor Pro calculate the x position of a 'Fit to Content' dropdown (ie: not full width) incorrectly.  This is because the element referenced when doing the `getBoundingClientRect()` calculations (in the `getMenuItemContainerAbsolutePosition()` function), it's using the toggle element to get the `left` property instead of its parent item element position.

**Content Settings in Menu Widget**
![content settings](https://github.com/elementor/elementor/assets/5360822/52435b46-cc76-4ddf-9083-8cfc8b8bf923)

**Layout example of current calculation**
![actual alignment](https://github.com/elementor/elementor/assets/5360822/19ffb6d9-b8a7-4dfc-ab84-cc8ff9c0de2b)

**Expected layout calculation**
![expected alignment](https://github.com/elementor/elementor/assets/5360822/f04a6969-970b-49d1-896b-b5cf7a148626)

**Element being used for calculations highlighted**
![using this element for calcs](https://github.com/elementor/elementor/assets/5360822/78877025-0ec0-4f24-a429-5702d48131ef)

**Element that was used for calculations before v3.16.0 highlighted**
![instead should use this element](https://github.com/elementor/elementor/assets/5360822/4015f990-7308-45a3-bead-31bec91c01d7)

Settings for 'Content Horizontal Position' that are 'left' or 'center' are effected. 'right' appears as expected as the toggle element is on the right side of the menu item.  

### Steps to reproduce

1. Add the Elementor Menu Widget to a page / header / template / etc.
![elementor menu](https://github.com/elementor/elementor/assets/5360822/428b7356-70e8-42ae-9296-c9bb31e8137c)

2. Set the 'Content Width' to 'Fit To Content' and 'Content Horizontal Position' to 'left'.
![settings to set](https://github.com/elementor/elementor/assets/5360822/552ea287-5db7-420d-85a0-401da2c78830)

3. Enable the dropdown for one of the menu items.
![enable dropdown](https://github.com/elementor/elementor/assets/5360822/f1f50a86-1d33-4fad-9c59-2c29c3741a57)

4. Add content to the dropdown.  It can be any widget.  I used the Text Widget in my examples.

5. Publish and view your menu.  Hover over the main menu item to show the dropdown. 

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt


					
== Server Environment ==
	Operating System: WINNT
	Software: Apache/2.4.57 (Win64) PHP/8.2.6
	MySQL version: MySQL Community Server (GPL) v5.7.26
	PHP Version: 8.2.6
	PHP Memory Limit: 600M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 500M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.3.1
	Site URL: http://local.mytestsite.com.au
	Home URL: http://local.mytestsite.com.au
	WP Multisite: No
	Max Upload Size: 500 MB
	Memory limit: 40M
	Max Memory limit: 600M
	Permalink Structure: /%postname%/
	Language: en-AU
	Timezone: Australia/Adelaide
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.8.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_AU
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.16.3
		Author: Elementor.com

	Elementor Pro
		Version: 3.16.1
		Author: Elementor.com


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	
	google_maps: Active


== Elementor Experiments ==
	Optimised DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Active by default
	Grid Container: Inactive by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Editor Top Bar: Inactive by default
	Landing Pages: Active by default
	Nested Elements: Active
	Lazy Load Background Images: Inactive by default
	Global Style Guide: Inactive by default
	Page Transitions: Active by default
	Notes: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Active
	Taxonomy Filter: Inactive by default


== Log ==
	
PHP: showing 6 of 6PHP: 2023-09-13 15:13:09 [notice X 1][..\wp-content\plugins\elementor\includes\utils.php::727] strtolower(): Passing null to parameter #1 ($string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-09-13 15:36:10 [notice X 48][..\wp-content\plugins\elementor\includes\elements\container.php::43] Creation of dynamic property Elementor\Includes\Elements\Container::$logical_dimensions_inline_end is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-09-15 14:42:21 [warning X 1][..\wp-content\plugins\elementor\core\common\modules\ajax\module.php::175] Undefined array key "data" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-09-15 14:42:23 [warning X 10][..\wp-content\plugins\elementor-pro\core\app\modules\site-editor\data\endpoints\templates.php::150] Undefined array key "condition_type" [array (
  'trace' => '
#0: ..\wp-content\plugins\elementor-pro\core\app\modules\site-editor\data\endpoints\templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: ..\wp-content\plugins\elementor-pro\core\app\modules\site-editor\data\endpoints\templates.php(120): class type array_map()
#3: ..\wp-content\plugins\elementor-pro\core\app\modules\site-editor\data\endpoints\templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: ..\wp-content\plugins\elementor\data\base\endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2023-09-15 14:42:30 [notice X 2][..\wp-content\plugins\elementor\includes\base\controls-stack.php::228] hexdec(): Passing null to parameter #1 ($hex_string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-09-15 15:35:20 [notice X 5][..\wp-content\plugins\elementor-pro\core\editor\editor.php::118] Using ${var} in strings is deprecated, use {$var} instead [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

Log: showing 20 of 352023-09-13 15:14:53 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.3',
  'to' => '3.16.1',
)]
2023-09-14 13:18:58 [info] elementor::elementor_updater Started 
2023-09-14 13:18:58 [info] Elementor/Upgrades - _on_each_version Start  
2023-09-14 13:18:58 [info] Elementor/Upgrades - _on_each_version Finished 
2023-09-14 13:18:58 [info] Elementor/Upgrades - _v_3_16_0_container_updates Start  
2023-09-14 13:18:58 [info] Elementor/Upgrades - _v_3_16_0_container_updates Finished 
2023-09-14 13:18:58 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Start  
2023-09-14 13:18:58 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Finished 
2023-09-14 13:18:58 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.16.1',
  'to' => '3.16.2',
)]
2023-09-15 12:13:39 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.16.2',
  'to' => '3.16.3',
)]
2023-09-15 12:13:40 [info] elementor::elementor_updater Started 
2023-09-15 12:13:40 [info] Elementor/Upgrades - _on_each_version Start  
2023-09-15 12:13:40 [info] Elementor/Upgrades - _on_each_version Finished 
2023-09-15 12:13:40 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Start  
2023-09-15 12:13:40 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Finished 
2023-09-15 12:13:40 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.16.2',
  'to' => '3.16.3',
)]
2023-09-15 12:14:35 [info] elementor-pro::elementor_pro_updater Started 
2023-09-15 12:14:35 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-09-15 12:14:35 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-09-15 12:14:35 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.16.0',
  'to' => '3.16.1',
)]

JS: showing 1 of 1JS: 2023-09-15 05:14:05 [error X 2][http://local.mytestsite.com.au/wp-content/plugins/elementor/assets/js/4fdaa70e951ad90db2f2.bundle.min.js:2:7786] Cannot read properties of null (reading 'dataset') 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	

				

````

</details>

**Comment by nicholaszein** [comment]
Hello @lizsterine! 

Thank you for submitting your issue! 🙏 

Please allow us some time to investigate it, and get back to you with our findings. 🧐 

Best regards

**Comment by nicholaszein** [comment]
Hello there @lizsterine!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.16.2! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
